orangecontrib/snom/preprocess/utils.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+
+from Orange.data import Domain
+from Orange.preprocess import Preprocess
+from orangecontrib.spectroscopy.preprocess import CommonDomain, SelectColumn
+from orangecontrib.spectroscopy.utils import (
+    InvalidAxisException,
+    values_to_linspace,
+    index_values,
+)
+
+
+class PreprocessImageOpts(Preprocess):
+    pass
+
+
+class PreprocessImageOpts2D(PreprocessImageOpts):
+    def __call__(self, data, image_opts):
+        common = self.image_transformer(data, image_opts)
+        at = data.domain[image_opts["attr_value"]].copy(
+            compute_value=SelectColumn(0, common)
+        )
+        domain = domain_with_single_attribute_in_x(at, data.domain)
+        return data.transform(domain)
+
+    def image_transformer(self, data, image_opts):
+        raise NotImplementedError
+
+
+def axes_to_ndim_linspace(coordinates):
+    # modified to avoid domains as much as possible
+    ls = []
+    indices = []
+
+    for i in range(coordinates.shape[1]):
+        coor = coordinates[:, i]
+        lsa = values_to_linspace(coor)
+        if lsa is None:
+            raise InvalidAxisException(i)
+        ls.append(lsa)
+        indices.append(index_values(coor, lsa))
+
+    return ls, tuple(indices)
+
+
+def get_ndim_hyperspec(data, attrs):
+    # mostly copied from orangecontrib.spectroscopy.utils,
+    # but returns the indices too
+    # also avoid Orange domains as much as possible
```

Comment:
I'm curious about this, is it faster? Should we just/also change the orangecontrib.spectroscopy.utils implementation? I think it used domains because `get_column()` didn't exist at that time.

Reply:
I did it because it seemed nicer to avoid domains where there are not needed. I do not expect any performance speedups, and yes, we could also do it in Spectroscopy.